### ISSUE TITLE
Telemetry 

### ISSUE THREAD

**Issue by Lalbabu2001** [rizacelik/STM32F411CEU6_INAV_Firmware#39]
I wanted to know does this flight controller support telemetry output?

**Comment by rizacelik** [comment]
You can use SOFTSERIAL for telemetry. You can turn it on from PORT settings. You may also need to configure it via CLI. You can learn the details by looking at INAV documents.
The pins you need are defined as follows. This is valid for quadquadcopter.

```
SOFTSERIAL TX PIN     PA0
SOFTSERIAL RX PIN     PA1
```

**Comment by Harshhhh16** [comment]
hello i used 6.1.1inav version but where to connect telemetry ?? i don't find serial port can you please tell me where to connect ??

**Comment by rizacelik** [comment]
What is your telemetry device?
The serial connection in INAV6.1.1 is as follows.

```
SOFTSERIAL TX PIN     PA0
SOFTSERIAL RX PIN     PA1

```

Why don't you switch to INAV 7.0 or INAV 7.1? you should do this. To benefit from the improvements, you must upgrade to INAV 7.1.

Support INAV 7.1 Serial

```
SOFTSERIAL TX PIN     PA9   // Softserial1_TX
SOFTSERIAL RX PIN     PA10  // Softserial1_RX

SOFTSERIAL TX PIN     PA0  // Softserial2_TX
SOFTSERIAL RX PIN     PA1 // Softserial2_RX
```

**Comment by Harshhhh16** [comment]
as your'e recommendation i switch to latest version but which is better accelerometer ? BMI160 vs MPU9250 vs MPU6500 ??

**Comment by rizacelik** [comment]
they are all good

**Comment by Harshhhh16** [comment]
hello i connect properly A0 And A1 pin to telemetry TX & RX but there is no data receive can you solve the problem ???



![Screenshot 2024-05-16 124644](https://github.com/rizacelik/STM32F411CEU6_INAV_Firmware/assets/166734051/eb8dd644-cd1e-4a0f-9d3d-5cfaf5287b1f)


is this correct ??

**Comment by rizacelik** [comment]
The answer I wrote to you is above. There is the following information there. It says which pins to use for which serial.

```
SOFTSERIAL TX PIN     PA9   // Softserial1_TX
SOFTSERIAL RX PIN     PA10  // Softserial1_RX

SOFTSERIAL TX PIN     PA0  // Softserial2_TX
SOFTSERIAL RX PIN     PA1 // Softserial2_RX
```

After testing, write the result here. Others will also benefit from this information and feedback.

**Comment by Harshhhh16** [comment]
All connection is same as you mentioned above 
I connect gps to PA9 PA10 its work
But Telemetry in PA0 PA1 is not working 
UART 1 is for gps 
UART 2 is for sbus
Soft serial 1 is for telemetry 
I’m right?

**Comment by rizacelik** [comment]
Have you looked at the diagram? Which pins is GPS connected to?
Use the following pins for telemetry.

```
SOFTSERIAL TX PIN PA9 // Softserial1_TX
SOFTSERIAL RX PIN PA10 // Softserial1_RX
```

If these pins do not work then install inav_7.1.0_STM32F411CE.zip.
Use the pins below when you upload this file.

```
SOFTSERIAL TX PIN PA0 // Softserial2_TX
SOFTSERIAL RX PIN PA1 // Softserial2_RX
```

**Comment by Harshhhh16** [comment]
![IMG_7347](https://github.com/rizacelik/STM32F411CEU6_INAV_Firmware/assets/166734051/88b679aa-383c-449e-8fbf-d9641f4bd46b)


I prefer this connection is it fine or I want to change it ?

**Comment by Harshhhh16** [comment]
Hello Now It's Work Properly i connect telemetry to PA9 & PA10
but in telemetry output is show 25v battery or 34v battery my battery is 3s 2200mah and also i connect resistor that is i didn't get 62k resistor so i connect 51k+10k+1k in series and 22k resistor still it's not measure correct voltage what's the problem ??

**Comment by rizacelik** [comment]
Which telemetry module are you using?

**Comment by Harshhhh16** [comment]
I saw a video on youtube that he makes hc12 module telemetry and alternate way is by hc05 module he also build with stm32 f411 from your github page his telemetry working fine but i think in my telemetry also  all parameters working its show altitude,navigation fine buy only battery voltage is incorrect…!

**Comment by rizacelik** [comment]
There will be a 62k resistor on the positive side of the battery. Watch out for this.

**Comment by rizacelik** [comment]
You can try the calculation below.

![image](https://github.com/rizacelik/STM32F411CEU6_INAV_Firmware/assets/19993109/b59a09d0-b0ce-4e35-87eb-6759febbe81b)

**Comment by Harshhhh16** [comment]
Yess i follow diagram 62k on n and 22k on negative terminal of battery and rest terminal common and goes to port B0 is I need to change 62k resistor to 14k ?

**Comment by rizacelik** [comment]
No. Use the red colored resistor value given in the diagram above.
Do not connect incorrectly. Board may burn.

**Comment by rizacelik** [comment]
![image](https://github.com/rizacelik/STM32F411CEU6_INAV_Firmware/assets/19993109/708d03af-0732-4f92-a94a-221da3c27b47)

**Comment by Harshhhh16** [comment]
Ok i go with this diagram let’s see if it’s work

**Comment by rizacelik** [comment]
Were you able to do it?

**Comment by Harshhhh16** [comment]
i recently burn out the board may be for wrong connection and also vl53lox sonar sensor also detected but it's need any calibration or set any mode ?? or it just plug and play
and second is how to setup ws2818 led strip ?? i don't understand it
rest all are working fine that's very nice and also thanks to you

**Comment by AaronSmith69** [comment]
> I saw a video on youtube that he makes hc12 module telemetry and alternate way is by hc05 module he also build with stm32 f411 from your github page his telemetry working fine but i think in my telemetry also all parameters working its show altitude,navigation fine buy only battery voltage is incorrect…!

do you have a video of that? i tried to build telemetry with hc-12 but it didn't get any info back to my GCS.
